pkg/provisioners/channel_util_test.go

```diff
@@ -0,0 +1,198 @@
+package provisioners
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/knative/pkg/apis"
+	istiov1alpha3 "github.com/knative/pkg/apis/istio/v1alpha3"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+)
+
+const (
+	channelName = "test-channel"
+	testNS      = "test-namespace"
+)
+
+var (
+	truePointer = true
+)
+
+func init() {
+	// Add types to scheme.
+	istiov1alpha3.AddToScheme(scheme.Scheme)
+	eventingv1alpha1.AddToScheme(scheme.Scheme)
+}
+
+func TestCreateK8sService(t *testing.T) {
```

Comment:
My personal preference is table tests, but I'm not sure if that is something the repo has standardized on.

Reply:
+1 on table tests for this situation where it is the same test body, but I would accept a followup for that refactor.

Reply:
Sure, I will do. Opened up #572 to track it